compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1310,7 +1310,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         visitor.visit_body(&body);
 
         let typeck_results = self.in_progress_typeck_results.map(|t| t.borrow()).unwrap();
-        let Some(liberated_sig) = typeck_results.liberated_fn_sigs().get(fn_hir_id) else { return false; };
+        let Some(liberated_sig) = typeck_results.liberated_fn_sigs().get(fn_hir_id).copied() else { return false; };
```

Comment:
This line changed because now the temporary lifetime rule applies. `typeck_results.liberated_fn_sigs()` loses extended temporary lifetime now, which is instead restricted to only the scope of this `let...else` binding statement.